File: yandex_code_life_balance/2. Узник замка Иф/solution.py
import sys
from typing import List

def can_fit(a: int, b: int, c: int, d: int, e: int) -> str:
    sizes = sorted([a, b, c])
    return "YES" if (sizes[0] <= d and sizes[1] <= e) or (sizes[0] <= e and sizes[1] <= d) else "NO"

def main():
    a: int = int(input())
    b: int = int(input())
    c: int = int(input())
    d: int = int(input())
    e: int = int(input())
    print(can_fit(a, b, c, d, e))

if __name__ == '__main__':
    main()

File: yandex_code_life_balance/2. Узник замка Иф/solution.cs
using System;

class Program 
{
    static string CanFit(int a, int b, int c, int d, int e)
    {
        int[] sizes = new int[3]{ a, b, c };
        Array.Sort(sizes);
        if ((sizes[0] <= d && sizes[1] <= e) ||  (sizes[0] <= e && sizes[1] <= d))
            return "YES";
        else
            return "NO";
    }

    static void Main() 
    {
        int a = int.Parse(Console.ReadLine());
        int b = int.Parse(Console.ReadLine());
        int c = int.Parse(Console.ReadLine());
        int d = int.Parse(Console.ReadLine());
        int e = int.Parse(Console.ReadLine());
        Console.WriteLine(CanFit(a, b, c, d, e));
    }
}